pkg/kando/location.go

```diff
@@ -38,17 +46,85 @@ func newLocationCommand() *cobra.Command {
 	cmd.AddCommand(newLocationDeleteCommand())
 	cmd.PersistentFlags().StringP(pathFlagName, "s", "", "Specify a path suffix (optional)")
 	cmd.PersistentFlags().StringP(profileFlagName, "p", "", "Pass a Profile as a JSON string (required)")
-	_ = cmd.MarkFlagRequired(profileFlagName)
+	cmd.PersistentFlags().StringP(repositoryServerFlagName, "r", "", "Pass a Repository Server CR as a JSON string (required for kopia based blueprints)")
 	return cmd
 }
 
+// NewDataMover creates an instance of DataMover Interface and returns
+// the preferred DataMover as per the arguments passed in kando command
+func NewDataMover(dm *DataMoverVal, outputName, snapJson string) datamover.DataMover {
+	switch dm.Type {
+	case profileFlagName:
+		return &datamover.Profile{
+			OutputName: outputName,
+			Profile:    dm.Profile,
+			SnapJSON:   snapJson,
+		}
+	case repositoryServerFlagName:
+		return &datamover.RepositoryServer{
+			OutputName:       outputName,
+			RepositoryServer: dm.RepositoryServer,
+			SnapJSON:         snapJson,
+		}
+	default:
+		return nil
+	}
+}
+
 func pathFlag(cmd *cobra.Command) string {
 	return cmd.Flag(pathFlagName).Value.String()
 }
 
+// validateCommandArgs makes sure that we are getting exactly
+// one of --profile or --repository-server flags
+func validateCommandArgs(cmd *cobra.Command) error {
+	profileFlag := cmd.Flags().Lookup(profileFlagName).Value.String()
+	repositoryServerFlag := cmd.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if profileFlag != "" && repositoryServerFlag != "" {
+		return errors.New("Either Provide --profile or --repository-server")
+	}
+	if profileFlag == "" && repositoryServerFlag == "" {
+		return errors.New("Please Provide either --profile or --repository-server as per the datamover you want to use")
+	}
+	return nil
+}
+
+func checkDataMover(cmd *cobra.Command) *DataMoverVal {
```

Comment:
I think this should be called `dataMoverFromCMD` instead. Also, instead of returning `*DataMoverVal` should we return  `(*DataMoverVal, error)` instead of that instead of returning empty struct we can return error, and handling error scenarios would be better in that case?

Reply:
Done !

Reply:
@r4rajat I still see that we are returning empty struct in error scenarios, can we please that like it's mentioned in the main review here.

Reply:
@viveksinghggits, I think the empty struct in error scenerio won't be causing any issues. As we're checking everytime for the error whenever we're calling this function.

Reply:
ok, cool.